File: articles/methods-in-moral-psychology-experiment-development-and-measurement/go/measurement_score.go
// Synthetic measurement score example for Methods in Moral Psychology.
// Educational scaffold only.
//
// Run:
//   go run go/measurement_score.go

package main

import "fmt"

func main() {
	indicators := map[string]float64{
		"wrongness":       0.8,
		"blame":           0.6,
		"norm_learning":   0.7,
		"action_tendency": 0.5,
	}

	weights := map[string]float64{
		"wrongness":       0.35,
		"blame":           0.30,
		"norm_learning":   0.20,
		"action_tendency": 0.15,
	}

	score := 0.0
	for variable, value := range indicators {
		score += value * weights[variable]
	}

	fmt.Printf("Synthetic latent moral construct score: %.3f\n", score)
}

File: articles/methods-in-moral-psychology-experiment-development-and-measurement/cpp/measurement_score.cpp
/*
Synthetic measurement score example for Methods in Moral Psychology.
Educational scaffold only.

Compile:
  c++ -std=c++17 cpp/measurement_score.cpp -o outputs/tables/measurement_score_cpp
Run:
  ./outputs/tables/measurement_score_cpp
*/

#include <iostream>
#include <map>
#include <string>

int main() {
    std::map<std::string, double> indicators = {
        {"wrongness", 0.8},
        {"blame", 0.6},
        {"norm_learning", 0.7},
        {"action_tendency", 0.5}
    };

    std::map<std::string, double> weights = {
        {"wrongness", 0.35},
        {"blame", 0.30},
        {"norm_learning", 0.20},
        {"action_tendency", 0.15}
    };

    double score = 0.0;

    for (const auto& item : indicators) {
        score += item.second * weights[item.first];
    }

    std::cout << "Synthetic latent moral construct score: "
              << score << std::endl;

    return 0;
}
